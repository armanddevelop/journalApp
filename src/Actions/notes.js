import { db } from "../FireBase/fireBaseConfig";
import { types } from "../Types/types";

export const notesActiveAction = (id, note) => ({
  type: types.notesActive,
  payload: {
    id,
    ...note,
  },
});

export const startNewNoteAction = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const newNote = {
      title: "",
      body: "",
      date: new Date().getTime(),
    };
    const docRef = await db.collection(`${uid}/jornal/notes`).add(newNote);
    dispatch(notesActiveAction(docRef.id, newNote));
  };
};
