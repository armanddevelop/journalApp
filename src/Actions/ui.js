import { types } from "../Types/types";

export const setErrorAction = (error) => ({
  type: types.uiSetError,
  payload: error,
});

export const unSetErrorAction = () => ({
  type: types.uiRemoveError,
});

export const startLoading = () => ({
  type: types.uiStartLoading,
});

export const finishLoading = () => ({
  type: types.uiFinishLoading,
});
