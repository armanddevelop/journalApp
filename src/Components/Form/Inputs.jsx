import PropTypes from "prop-types";
export const Inputs = ({ page }) => {
  return (
    <>
      {page === "login" ? (
        <>
          <input
            className="auth__input"
            type="text"
            placeholder="Email"
            name="email"
            autoComplete="off"
          />
          <input
            className="auth__input"
            type="password"
            placeholder="password"
            name="password"
          />
        </>
      ) : (
        <>
          <input
            className="auth__input"
            type="text"
            placeholder="Name"
            name="name"
            autoComplete="off"
          />
          <input
            className="auth__input"
            type="text"
            placeholder="Email"
            name="email"
            autoComplete="off"
          />
          <input
            className="auth__input"
            type="password"
            placeholder="password"
            name="password"
          />
          <input
            className="auth__input"
            type="password"
            placeholder="Confirm password"
            name="confirmPassword"
          />
        </>
      )}
    </>
  );
};
Inputs.propTypes = {
  page: PropTypes.string.isRequired,
};
