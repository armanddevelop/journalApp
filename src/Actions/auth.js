import { types } from "../Types/types";
import { firebase, googleAuthProvider } from "../FireBase/fireBaseConfig";
import {
  finishLoading,
  startLoading,
  errorInFirebase,
  removeErrorInFirebase,
} from "./ui";
import { notesLogOutAction } from "./notes";

export const logInAction = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});

export const logOutAction = () => ({
  type: types.logout,
});

export const startLogOut = () => {
  return async (dispatch) => {
    try {
      await firebase.auth().signOut();
      dispatch(logOutAction());
      dispatch(notesLogOutAction());
    } catch (error) {
      console.log("shit happen in startLogOut ", error);
    }
  };
};

export const getGoogleLogInAction = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
      });
  };
};

export const getLoginUserAction = (email, password) => {
  return (dispatch) => {
    dispatch(startLoading());
    dispatch(removeErrorInFirebase());
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
        dispatch(finishLoading());
      })
      .catch((error) => {
        dispatch(finishLoading());
        dispatch(errorInFirebase(error.message));
      });
  };
};

export const registerUserAction = (name, email, password) => {
  return (dispatch) => {
    dispatch(startLoading());
    dispatch(removeErrorInFirebase());
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
        dispatch(finishLoading());
      })
      .catch((error) => {
        dispatch(finishLoading());
        dispatch(errorInFirebase(error.message));
      });
  };
};
