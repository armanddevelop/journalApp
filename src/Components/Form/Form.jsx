import { Inputs } from "./Inputs";
import PropTypes from "prop-types";
export const Form = ({ pageName, page }) => {
  return (
    <form>
      <Inputs page={page} />
      <button className="btn btn-primary btn-block mb-5" type="submit">
        {pageName}
      </button>
    </form>
  );
};

Form.propTypes = {
  pageName: PropTypes.string.isRequired,
  page: PropTypes.string.isRequired,
};
