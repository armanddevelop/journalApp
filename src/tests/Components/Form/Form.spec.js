import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { getLoginUserAction, registerUserAction } from "../../../Actions/auth";
import { Form } from "../../../Components/Form/Form";

jest.mock("../../../Actions/auth", () => ({
  getLoginUserAction: jest.fn(),
  registerUserAction: jest.fn(),
}));
jest.mock("../../../Components/Utils/validations", () => ({
  isFormValidRegister: () => true,
  isFormValidLogin: () => true,
}));
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
  auth: {},
  ui: {
    msgError: "",
    loading: false,
  },
};
let store = mockStore(initState);
store.dispatch = jest.fn();
const values = {
  email: "test@hotmail.com",
  password: "als811",
  name: "alicha",
};
describe("Test in Form component login form", () => {
  const wrapper = mount(
    <Provider store={store}>
      <Form
        pageName="Login"
        page="login"
        values={values}
        handleInputChange={() => {}}
      />
    </Provider>
  );

  it("SnapShot to Form component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should mount login form", () => {
    const content = wrapper.find("button").text();

    expect(content).toEqual("Login");
  });

  it("Should show only two inputs", () => {
    const inputs = wrapper.find("input").length;

    expect(inputs).toEqual(2);
  });

  it("should execute getLoginUserAction", () => {
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    expect(getLoginUserAction).toHaveBeenCalledWith(
      values.email,
      values.password
    );
  });

  describe("Test in Form component register form", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Form
          pageName="Register"
          page="register"
          values={values}
          handleInputChange={() => {}}
        />
      </Provider>
    );

    it("To match Snapshot register", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("Should show the inputs in register", () => {
      const inputs = wrapper.find("input").length;

      expect(inputs).toEqual(4);
    });

    it("should execute registerUserAction", () => {
      wrapper.find("form").simulate("submit", { preventDefault: () => {} });

      expect(registerUserAction).toHaveBeenCalledWith(
        values.name,
        values.email,
        values.password
      );
    });
  });
  describe("Snapshot with the AlertUI component", () => {
    const initState = {
      auth: {},
      ui: {
        msgError: "shit happend in Form",
        loading: true,
      },
    };
    let store = mockStore(initState);
    const wrapper = mount(
      <Provider store={store}>
        <Form
          pageName="Register"
          page="register"
          values={values}
          handleInputChange={() => {}}
        />
      </Provider>
    );

    it("Should show AlertUI componentn in snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("Should have alerts__container in AlertUI component", () => {
      const propertie = wrapper.find(".alerts__container").exists();

      expect(propertie).toEqual(true);
    });

    it("Should have the error message in AlertUI component", () => {
      const msgError = wrapper.find(".MuiAlert-message").first().text();

      expect(msgError).toEqual(initState.ui.msgError);
    });
  });
});
