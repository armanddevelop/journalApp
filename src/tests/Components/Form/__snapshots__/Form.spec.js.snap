// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Form component login form SnapShot to Form component 1`] = `
<form
  className="animate__animated animate__fadeIn"
  onSubmit={[Function]}
>
  Array [
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="test@hotmail.com"
    />,
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value="als811"
    />,
  ]
  <button
    className="btn btn-primary btn-block mb-5"
    disabled={false}
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`Test in Form component login form Snapshot with the AlertUI component Should show AlertUI componentn in snapshot 1`] = `
<form
  className="animate__animated animate__fadeIn"
  onSubmit={[Function]}
>
  <div
    className="alerts__container"
  >
    Array [
      "",
      Array [
        "",
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-xilmwd-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          Array [
            "",
            <div
              className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              Array [
                "",
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>,
              ]
            </div>,
          ]
          Array [
            "",
            <div
              className="MuiAlert-message css-acap47-MuiAlert-message"
            >
              shit happend in Form
            </div>,
          ]
        </div>,
      ],
    ]
  </div>
  Array [
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="alicha"
    />,
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="test@hotmail.com"
    />,
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value="als811"
    />,
    <input
      className="auth__input"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
    />,
  ]
  <button
    className="btn btn-primary btn-block mb-5"
    disabled={true}
    type="submit"
  >
    Register
  </button>
</form>
`;

exports[`Test in Form component login form Test in Form component register form To match Snapshot register 1`] = `
<form
  className="animate__animated animate__fadeIn"
  onSubmit={[Function]}
>
  Array [
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="alicha"
    />,
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="test@hotmail.com"
    />,
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value="als811"
    />,
    <input
      className="auth__input"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
    />,
  ]
  <button
    className="btn btn-primary btn-block mb-5"
    disabled={false}
    type="submit"
  >
    Register
  </button>
</form>
`;
